data_structures/queue/fifo_animal_shelter.py

```diff
@@ -17,9 +50,9 @@ def __repr__(self):
 
     def enqueue(self, animal=None):
         """Insert an animal to the back of the queue"""
-        node = Node(animal)
+        node = Node(animal.species)
 
-        if animal != 'dog' and animal != 'cat':
+        if animal.species != 'dog' and animal.species != 'cat':
```

Comment:
You also need to change your `dequeue()` method to check the `species` of the animal like this. `current.val == pref` will never be `True` since `current.val` is some animal object and `pref` is a string.